presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -530,6 +588,30 @@ protected void invalidateTable(String databaseName, String tableName)
             String tableName,
             Map<Column, Domain> partitionPredicates)
     {
+        if (isPartitionVersioningEnabled) {
+            List<String> result = new ArrayList<>();
+            delegate.getPartitionNamesWithVersionByFilter(databaseName, tableName, partitionPredicates).forEach(partitionName -> {
```

Comment:
Overall I find this `forEach` a bit too complicated. Maybe something like the following might be better:

```
List<PartitionNameWithVersion> partitionNamesWithVersion = delegate.getPartitionNamesWithVersionByFilter(databaseName, tableName, partitionPredicates);
List<String> result = partitionNamesWithVersion....collect(...);
partitionNamesWithVersion.forEach(this::invalidateStalePartition); // Or invalidateStalePartitions(partitionNamesWithVersion)
```

Otherwise, we should probably just use a for loop.